src/RoslynPad.Common.UI/ViewModels/MainViewModelBase.cs

```diff
@@ -461,6 +461,10 @@ public string SearchText
             set
             {
                 SetProperty(ref _searchText, value);
+                if (Settings.SearchWhileTyping)
```

Comment:
Add `if (SetProperty(...)` - it would prevent the same value from triggering the command again needlessly.